source/views/settings/screens/licenses/index.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import * as React from 'react'
+import {View, StyleSheet} from 'react-native'
+import {LicensesList} from './list'
+import {type LicenseType} from './types'
+import {type NavigationScreenProp} from 'react-navigation'
+
+import licenseData from '../../../../../docs/licenses.json'
+
+type Props = {
+	navigation: NavigationScreenProp<*>,
+}
+
+export class LicensesView extends React.Component<Props> {
+	static navigationOptions = {
+		title: 'Open Source Licenses',
+	}
+
+	extractNameFromGithubUrl = (url: string) => {
+		if (!url) {
+			return null
+		}
+
+		const reg = /((https?:\/\/)?(www\.)?github\.com\/)?(@|#!\/)?([A-Za-z0-9_]{1,15})(\/([-a-z]{1,20}))?/iu
```

Comment:
According to the form validation parameters on [GitHub's signup page](https://github.com/join), the parameters for a username are

- A username may only contain alphanumeric characters or hyphens.
- A username cannot have multiple consecutive hyphens.
- A username cannot begin or end with a hyphen.
- Usernames have a maximum length of 39 characters.

There are similar constraints somewhere for the repository name.

All of this is to say that this regex doesn't match the following types of GitHub urls:

- URLs with a username component containing hyphens. (e.g. https://github.com/frog-pond/ccc-server)
- URLs with a repository component containing numbers. (e.g. https://github.com/rye/rust-netaddr2)

Reply:
Thank you for listing the form validation parameters. I think a few things could use tweaking seeing your feedback.

1. We can add capturing for hyphens in the username
1. We can add capturing for numbers in the repo name
1. We can completely ignore max length of username

What are your thoughts on something like this, then?

```diff
-([A-Za-z0-9_]{1,15})(\/([-a-z]{1,20}))
+([A-Za-z0-9_-]{1,})(\/([-a-z0-9]{1,}))
```

* Here's a [repl.it](https://repl.it/@drewvolz/TriangularDistinctQuadrants) to demonstrate the changes to the username capturing
* Here's a [regex101](https://regex101.com/r/AKwSjV/1) to demo all captured parts